src/PAModel/Serializers/MsAppSerializer.cs

```diff
@@ -537,14 +537,22 @@ private static IEnumerable<FileEntry> GetMsAppFiles(this CanvasDocument app, Err
                 yield return ToFile(FileKind.PublishInfo, publishInfo);
 
             // "DataComponent" data sources are not part of DataSource.json, and instead in their own file
+            // In case the the entropy is missing then make sure that tableDefinitions are being added towards the end in DataSources.json file
+            // This is to respect the serialization order of the server.
             var dataSources = new DataSourcesJson
             {
                 DataSources = app.GetDataSources()
                     .SelectMany(x => x.Value)
-                    .Where(x => !x.IsDataComponent)
+                    .Where(x => !x.IsDataComponent && x.TableDefinition == null)
                     .OrderBy(x => app._entropy.GetOrder(x))
```

Comment:
>GetOrder [](http://example.com/codeflow?start=47&length=8)

Can we just make GetOrder() be "smarter", ie when there's no entropy , it can put TableDefinitions at the end. 

There's a potential issue here: this could potentially reorder from what's in entropy.

Reply:
Ie, something like:
        public int GetOrder(DataSourceEntry dataSource)
        {            
            return this.OrderDataSource.GetOrDefault<string, int>(dataSource.GetUniqueName(),
                (dataSource.TableDefinition == null) ? -1 : int.MaxValue
             );
        }